### DISCUSSION TITLE
SNES firmware receiver shifts the miser button sequence through SNAC

### DISCUSSION THREAD

**Discussion by Hifly-rc** [darthcloud/BlueRetro#1098] [General]
Thank you to the author for opening up the BR project. During my self-made process, I found that the SNES firmware receiver shifted the mister button sequence through SNAC, and the PS4 controller's up key was incorrectly recognized as the down key, and the controller's down key was recognized as the left key, which is equivalent to one clock cycle. I tried to add and remove the SN74 logic IC on the circuit, but it didn't work. The original SNES controller input SNAC is working normally. I don't know where to solve the problem, hardware or software?

**Comment by Hifly-rc** [comment]
![QQ20241016-171333](https://github.com/user-attachments/assets/1faf51a0-5044-4b07-9c03-6b2802220c37)
This is a logical sequence, the original SNES to SNAC control is correct, but the PS4 controller BR to SNAC control is offset due to errors

**Comment by darthcloud** [comment]
Your clock polarity is wrong, a real sness clock is idle high

**Comment by bumbec** [comment]
Exactly the same problem on murmulator pico-nes. How to fix this?

**Reply by Hifly-rc** [reply]
> Exactly the same problem on murmulator pico-nes. How to fix this?murmulator pico-nes 上的问题完全相同。怎么解决这个问题？

To be precise, I applied BR to the Analogizer project. The 8-bit Bluetooth controller works normally on the Analogizer, just like the original Nintendo SFC controller.

**Comment by bumbec** [comment]
This project is only suitable for original retro consoles.
It does not exactly emulate game controllers.
